paymentapi/src/test/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentControllerTest.java

```diff
@@ -0,0 +1,218 @@
+        });
+        Assertions.assertEquals(CommonApiException.class, exception1.getClass());
+        Assertions.assertEquals(HttpStatus.FORBIDDEN, exception1.getStatus());
+        Assertions.assertEquals("rejected-creating-paytrail-payment-for-unconfirmed-order", exception1.getErrors().getErrors().get(0).getCode());
+        Assertions.assertEquals("rejected creating paytrail payment for unconfirmed order, order id [" + orderWrapper1.getOrder().getOrderId() + "]", exception1.getErrors().getErrors().get(0).getMessage());
+
+        /* Test with OrderStatus.CANCELLED */
+        OrderWrapper orderWrapper2 = createDummyOrderWrapper();
+        orderWrapper2.getOrder().setStatus("cancelled");
+        paymentRequestDataDto.setOrder(orderWrapper2);
+
+        CommonApiException exception2 = assertThrows(CommonApiException.class, () -> {
+            paytrailPaymentController.createPaymentFromOrder(paymentRequestDataDto);
+        });
+        Assertions.assertEquals(CommonApiException.class, exception2.getClass());
+        Assertions.assertEquals(HttpStatus.FORBIDDEN, exception2.getStatus());
+        Assertions.assertEquals("rejected-creating-paytrail-payment-for-unconfirmed-order", exception2.getErrors().getErrors().get(0).getCode());
+        Assertions.assertEquals("rejected creating paytrail payment for unconfirmed order, order id [" + orderWrapper2.getOrder().getOrderId() + "]", exception2.getErrors().getErrors().get(0).getMessage());
+    }
+
+    private OrderWrapper createDummyOrderWrapper() {
+        OrderDto orderDto = new OrderDto();
+        String orderId = UUID.randomUUID().toString();
+        orderDto.setOrderId(orderId);
+        orderDto.setNamespace("venepaikat");
+        orderDto.setUser("dummy_user");
+        orderDto.setCreatedAt("");
+        orderDto.setStatus("confirmed");
+        orderDto.setType("order");
+        orderDto.setCustomerFirstName("Martin");
+        orderDto.setCustomerLastName("Leh");
+        orderDto.setCustomerEmail("testi@ambientia.fi");
+        orderDto.setPriceNet("1234");
+        orderDto.setPriceVat("0");
+        // Sets total price to be 1 eur
+        orderDto.setPriceTotal("1234");
+
+        OrderWrapper order = new OrderWrapper();
+        order.setOrder(orderDto);
+
+        List<OrderItemDto> items = new ArrayList<>();
+
+        OrderItemDto orderItem = new OrderItemDto();
+
+        String orderItemId = UUID.randomUUID().toString();
+        orderItem.setOrderItemId(orderItemId);
+        orderItem.setPriceGross(BigDecimal.valueOf(1234));
+        orderItem.setQuantity(1);
+        orderItem.setVatPercentage("24");
+        orderItem.setProductId("test-product-id");
+        orderItem.setProductName("productName");
+        orderItem.setOrderId(orderId);
+
+        items.add(orderItem);
+
+        order.setItems(items);
+        return order;
+    }
+
+    public String getFirstMerchantIdFromNamespace(String namespace) {
```

Comment:
Could you create TestUtil class like in [here](https://github.com/City-of-Helsinki/verkkokauppa-core/blob/07be80a2d8914bc31bbd5f915303788cbb13f70c/orderapi/src/test/java/fi/hel/verkkokauppa/order/test/utils/TestUtils.java#L46) to make this function usage easier